c10/util/UniqueVoidPtr.h

```diff
@@ -66,6 +66,11 @@ class UniqueVoidPtr {
   std::unique_ptr<void, DeleterFnPtr>&& move_context() {
     return std::move(ctx_);
   }
+  C10_NODISCARD bool compare_exchange_deleter(DeleterFnPtr expected_deleter, DeleterFnPtr new_deleter) {
+    if (get_deleter() != expected_deleter) return false;
```

Comment:
I associate the compare_exchange name with thread safety. Should this be thread safe?

Reply:
If you insist, we can call it `nonatomic_compare_exchange_deleter` :) (there is no reason to make it thread safe.)